sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -908,4 +909,25 @@ object JdbcUtils extends Logging {
       statement.close()
     }
   }
+
+  /**
+   * Update a table from the JDBC database.
+   */
+  def alterTable(
+      conn: Connection,
+      tableName: String,
+      changes: Seq[TableChange],
+      options: JDBCOptions
+      ): Unit = {
+    val dialect = JdbcDialects.get(options.url)
+    val statement = conn.createStatement
+    try {
+      statement.setQueryTimeout(options.queryTimeout)
+      for (sql <- dialect.alterTable(tableName, changes)) {
+        statement.executeUpdate(sql)
+      }
```

Comment:
I am debating if I should use `statement.executeBatch`. The code is simpler without using batch. Not sure if it is common for user to add lots of columns at one time.

Reply:
What happens if one of the statements fails? Do we leave the table in partially modified state? Should we perform all the statements atomically?

Reply:
Fixed